prowler/providers/gcp/gcp_provider.py

```diff
@@ -238,12 +243,46 @@ def setup_session(self, credentials_file: str, service_account: str) -> Credenti
             )
             sys.exit(1)
 
-    def __set_gcp_creds_env_var__(self, credentials_file):
-        logger.info(
-            "GCP provider: Setting GOOGLE_APPLICATION_CREDENTIALS environment variable..."
-        )
-        client_secrets_path = os.path.abspath(credentials_file)
-        os.environ["GOOGLE_APPLICATION_CREDENTIALS"] = client_secrets_path
+    @staticmethod
+    def test_connection(
+        credentials_file: str, service_account: str, raise_on_exception: bool = True
```

Comment:
Is it mandatory to set the `credentials_file` or `service_account`? If not please set some default values.